src/app/core/token-interceptor.service.ts

```diff
@@ -0,0 +1,24 @@
+import {Inject, Injectable} from '@angular/core';
+import {HttpEvent, HttpHandler, HttpRequest} from '@angular/common/http';
+import {Observable} from 'rxjs';
+import {AuthConfig, AuthTokenConfig} from '@core';
+
+@Injectable()
+export class TokenInterceptorService {
+
+  constructor(@Inject(AuthConfig) private config: AuthTokenConfig) {
+  }
+
+  intercept(request: HttpRequest<any>, next: HttpHandler): Observable<HttpEvent<any>> {
+    const token = localStorage.getItem('app_token');
+    if (!request.url.startsWith(this.config.ApiUrl)
```

Comment:
extract this logic inside an `isWhitelisted` `pure` method to improve code readability and make sure the code is written with functional programming in mind.